packages/web-react/src/components/Modal/ModalCloseButton.tsx

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable import/no-extraneous-dependencies, import/extensions, import/no-unresolved */
+import React, { createElement } from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore: No declaration file
+import icons from '@lmc-eu/spirit-icons/dist/icons';
+import { SpiritModalCloseButtonProps } from '../../types';
+import { IconsProvider } from '../../context';
+import { useClassNamePrefix } from '../../hooks';
+import { Icon } from '../Icon';
+
+const ModalCloseButton = (props: SpiritModalCloseButtonProps) => {
+  const { UNSAFE_style, UNSAFE_className, ...rest } = props;
+
+  const modalCloseClass = useClassNamePrefix('Modal__close');
+  const buttonProps = {
+    role: 'button',
+    className: [
+      'Button',
+      'Button--tertiary',
+      'Button--square',
+      'Button--medium',
+      modalCloseClass,
+      UNSAFE_className,
+    ].join(' '),
+    style: UNSAFE_style,
+    ...rest,
+  };
+
+  return createElement(
+    'button',
+    {
+      type: 'button',
+      ...buttonProps,
+    },
+    <>
+      <IconsProvider value={icons}>
```

Comment:
This does not belong to this place. Use Icon directly but Provider you must use in an example like Storybook. Icons are interchangeable and now you are fixing them to Spirit Icons only.